src/openms/source/APPLICATIONS/TOPPBase.cpp

```diff
@@ -387,6 +408,130 @@ namespace OpenMS
+        platform = "Win";
+      #elif __APPLE__
+        platform = "Mac";
+      #else
+        platform = "Linux";
+      #endif
+
+      String architecture = QSysInfo::WordSize == 32 ? "32" : "64";
+
+      // write to tmp + userid folder
+
+      // e.g.: OpenMS_Default_Win_64_FeatureFinderCentroided_2.0.0
+      String tool_version_string;
+      tool_version_string = String("OpenMS") + "_" + "Default_" + platform + "_" + architecture + "_" + tool_name_ + "_" + version_;
+
+      String version_file_name = File::getOpenMSHomePath() + "/.OpenMS/" + tool_name_ + ".ver";
+
+      // create version file if it doesn't exist yet
+      bool first_run(false);
+      if (!File::exists(version_file_name) || !File::readable(version_file_name))
+      {
+        Param p = File::getSystemParameters(); // initializes .OpenMS folder
+
+        // touch file to create it and set initial modification time stamp
+        QFile f;
+        f.setFileName(version_file_name.toQString());
+        f.open(QIODevice::WriteOnly);
+        f.close();
+        first_run = true;
+      }
+      
+      if (File::readable(version_file_name))
+      {
+        QDateTime last_modified_dt = QFileInfo(version_file_name.toQString()).lastModified();
+        QDateTime current_dt = QDateTime::currentDateTime();
+
+        // check if at least one day passed sincle last request
+        if (first_run || current_dt > last_modified_dt.addDays(1))
+        {
+          // update modification time stamp
+          struct stat old_stat;
+          time_t mtime;
+          struct utimbuf new_times;
+          stat(version_file_name.c_str(), &old_stat);
+          mtime = old_stat.st_mtime; 
+          new_times.actime = old_stat.st_atime; // keep accessuib time unchanged 
+          new_times.modtime = time(NULL);  // mod time to current time
+          utime(version_file_name.c_str(), &new_times);          
+
+          LOG_INFO << "The OpenMS team is collecting use statistics for quality control and funding purposes." << endl;
+          LOG_INFO << "We will never give out your personal data but you may disable this functionality by setting the environmental variable OPENMS_DISABLE_USAGE_STATISTICS." << endl;
+          
+          // Usage of a QCoreApplication is overkill here (and ugly too), but we just use the
+          // QEventLoop to process the signals and slots and grab the results afterwards from
+          // the MascotRemotQuery instance
+          char** argv2 = const_cast<char**>(argv);
+          QCoreApplication event_loop(argc, argv2);
+          NetworkGetRequest* query = new NetworkGetRequest(&event_loop);
+          query->setUrl(QUrl(QString("http://openms-update.informatik.uni-tuebingen.de/check/") + tool_version_string.toQString()));
+          QObject::connect(query, SIGNAL(done()), &event_loop, SLOT(quit()));
+          QTimer::singleShot(1000, query, SLOT(run()));          
+          QTimer::singleShot(5000, query, SLOT(timeOut()));
```

Comment:
It's only one attempt. The first one shot runs the query (run) and the second shot emits the timeout signal.